### PR TITLE
Set languages in alphabetic order

### PR DESCRIPTION
**Pull request by boutinb**

### CODE REVIEW

File: Desktop/utilities/languagemodel.cpp [hunk 1]
```diff
@@ -336,7 +331,9 @@ void LanguageModel::findQmFiles(QString qmlocation)
 		{
 			Log::log() << "Language (" << loc.language() << ") not registered in LanguageModel, adding it now" << std::endl;
 
-			_languages.push_back(loc.language());
+			int i = 0;
+			while (i < _languages.size() && loc.nativeLanguageName().compare(_languagesInfo[_languages[i]].nativeLanguageName, Qt::CaseInsensitive) > 0)	i++;
```
Comment (JorisGoosen) [inline Desktop/utilities/languagemodel.cpp]: Why are you using a while loop to add random empty languages to the list? 

Why not make _languages a more sensible datatype...

Comment (JorisGoosen) [reply Desktop/utilities/languagemodel.cpp]: Or just throw it away and use the keys as they are in _languagesInfo?

Comment (JorisGoosen) [reply Desktop/utilities/languagemodel.cpp]: In any case, make the while loop into a for loop.

Comment (boutinb) [reply Desktop/utilities/languagemodel.cpp]: I have restructured the LanguageModel class:

- There is no currentIndex anymore, only a languageName
- _languages map as for key the value displayed in the dropdown. This key is composed by the language code + the native name of the language. So the order of the language code will be used. And as the native language name is used, thanks to the language code, it is easier to know which language the name means (I have no clue to distinguish Japanese from Chinese).
- The locale country is also used: we can now differentiate a language spoken in different country (I know that the portuguese is now mainly Brazilan Portuguese, but in the future Portugal Portuguese could be added).
- Obsolete codes are removed.

File: Desktop/utilities/languagemodel.h [hunk 2]
```diff
@@ -45,71 +33,58 @@ class LanguageModel : public QAbstractListModel
...
-	QString currentLanguageCode()	const;
-	QLocale::Language currentLanguage()	const {return currentLanguageInfo().language; };
+	QString languageCode(const QLocale& locale)		const;
+	QString languageCode(const QString& language)	const;
+	QString	languageName(const QLocale& loc)		const;
+	QString currentLanguageCode()					const	{ return languageCode(_currentLanguage); }
+	QString currentLanguage()						const	{ return _currentLanguage; }
 
+	static	bool	hasDefaultLanguage()		{ return _singleton->currentLanguage() == _singleton->languageName(_defaultLocale); }
 	//This function (currentTranslationSuffix) should be made obsolete through the abolishment of all the _nl etc files:
```
Comment (JorisGoosen) [inline Desktop/utilities/languagemodel.h:51]: Amen!

File: Desktop/utilities/settings.cpp [hunk 3]
```diff
@@ -38,6 +38,7 @@ const Settings::Setting Settings::Values[] = {
 	{"cranRepositoryURL",			"https://cloud.r-project.org"},
 	{"userHasGitHubAccount",		false},
 	{"preferredLanguage",			QLocale::English},
+	{"preferredCountry",			QLocale::UnitedKingdom},
```
Comment (JorisGoosen) [inline Desktop/utilities/settings.cpp:41]: Would @EJWagenmakers prefer `QLocale::UnitedStates`?

**Review by JorisGoosen — APPROVED** [review]
That was a delight to review!

**Comment by boutinb** [comment]
> That was a delight to review!

Thanks!

**Pull request merged**